xbmc/utils/Signals.h

```diff
@@ -0,0 +1,201 @@
+/*
```

Comment:
I am fine with this new approach when kept private in the first round. Later we should start some consolidation and agree when to use what pattern. (still hoping that this will happen sometime :)